### PR TITLE
demisto/etc#23148: Wrapped vulnDB integration in base client 

### PR DESCRIPTION
**Pull request by hod-alpert**
## Status
- [ ] In Progress
- [x] Ready
- [ ] In Hold - (Reason for hold)

## Related Issues
fixes: demisto/etc#23148

## Description
Wrapped vulnDB integration in base client

### CODE REVIEW

**Comment by Itay4** [comment]
did we verify test playbook passes?

File: Integrations/VulnDB/VulnDB.py [hunk 1]
```diff
@@ -421,31 +379,51 @@ def vulndb_get_cve_command():
...
+    # Remove trailing slash to prevent wrong URL path to service
+    api_url = params['api_url'].rstrip('/')
+    client_id = params['client_id']
+    client_secret = params['client_secret']
+    use_ssl = not params.get('insecure', False)
+    proxy = params.get('proxy', False)
+    client = Client(proxy, use_ssl, api_url, client_id, client_secret)
+    args = demisto.args()
+    command = demisto.command()
+    LOG('Command being called is %s' % (demisto.command()))
```
Comment (Itay4) [inline Integrations/VulnDB/VulnDB.py:394]: lets use f-strings here

File: Integrations/VulnDB/VulnDB.py [hunk 2]
```diff
@@ -421,31 +379,51 @@ def vulndb_get_cve_command():
...
-    vulndb_get_version_command()
-elif demisto.command() == 'vulndb-get-updates-by-dates-or-hours':
-    vulndb_get_updates_by_dates_or_hours_command()
-elif demisto.command() == 'cve':
-    vulndb_get_cve_command()
+
+def main():
+    params = demisto.params()
+    # Remove trailing slash to prevent wrong URL path to service
+    api_url = params['api_url'].rstrip('/')
```
Comment (Itay4) [inline Integrations/VulnDB/VulnDB.py:386]: i think `BaseClient` already handles that

File: Integrations/VulnDB/VulnDB.py [hunk 3]
```diff
@@ -10,57 +10,42 @@
...
+        return res.get('access_token')
+
+    def http_request(self, url_suffix, size=None):
+        params = {'size': size} if size else None
+        res = self._http_request("GET",
+                                 url_suffix=url_suffix,
+                                 params=params)
+        for error_attribute in ['error', 'details']:
+            if error_attribute in res:
+                return_error(res[error_attribute])
```
Comment (Itay4) [inline Integrations/VulnDB/VulnDB.py:47]: raise exception instead of `return_error`

File: Integrations/VulnDB/VulnDB.py [hunk 4]
```diff
@@ -421,31 +379,51 @@ def vulndb_get_cve_command():
...
+        elif command == 'vulndb-get-product':
+            vulndb_get_product_command(args, client)
+        elif command == 'vulndb-get-version':
+            vulndb_get_version_command(args, client)
+        elif command == 'vulndb-get-updates-by-dates-or-hours':
+            vulndb_get_updates_by_dates_or_hours_command(args, client)
+        elif command == 'cve':
+            vulndb_get_cve_command(args, client)
+    except Exception as e:
+        error_message = str(e)
```
Comment (Itay4) [inline Integrations/VulnDB/VulnDB.py:424]:
```suggestion
        error_message = f'Failed to execute {command} command. Error: {str(e)}')
```

File: Integrations/VulnDB/VulnDB.py [hunk 5]
```diff
@@ -10,57 +10,42 @@
...
+                                 },
+                                 full_url=oath_url)
+        return res.get('access_token')
+
+    def http_request(self, url_suffix, size=None):
+        params = {'size': size} if size else None
+        res = self._http_request("GET",
+                                 url_suffix=url_suffix,
+                                 params=params)
+        for error_attribute in ['error', 'details']:
```
Comment (Itay4) [inline Integrations/VulnDB/VulnDB.py:45]: add comment here that its either one of the two

**Comment by hod-alpert** [comment]
@Itay4 Added unit tests and the playbook passes when there is quota, when there isn't it will display the correct error.

In addition- @ShahafBenYakir changed its configurations so that it will only run at nightly to avoid exceeding the limit

**Comment by kirbles19** [comment]
@hod-alpert Done.

**Pull request merged**